crates/storage/nippy-jar/src/compression/mod.rs

```diff
@@ -0,0 +1,66 @@
+use crate::NippyJarError;
+use serde::{Deserialize, Serialize};
+use std::io::Write;
+
+mod zstd;
+pub use zstd::{Zstd, ZstdState};
+
+pub trait Compression {
+    /// Returns decompressed data.
+    fn decompress(&self, value: &[u8]) -> Result<Vec<u8>, NippyJarError>;
+
+    /// Compresses data from `src` to `dest`
+    fn compress_to<W: Write>(&self, src: &[u8], dest: &mut W) -> Result<(), NippyJarError>;
+
+    /// Returns `true` if it's ready to compress.
+    fn is_ready(&self) -> bool {
+        true
+    }
+
+    /// If required, prepares compression algorithm with an early pass on the data.
+    fn prepare_compression(
+        &mut self,
+        _columns: Vec<impl IntoIterator<Item = Vec<u8>>>,
```

Comment:
given that index files can be pretty large, let's avoid cloning here and pass references instead? If I understand correctly, it'll eventually unroll into `&[u8]` anyway

Reply:
this is unrelated to the index files, but `Item` will probably be a AsRef kinda of thing